### PR TITLE
Adopt ceil method for ios inapp prices

### PR DESCRIPTION
**Pull request by jawad-khan**
# ⛔️ MAIN BRANCH WARNING! 2U EMPLOYEES must make branches against the 2u/main BRANCH

- [ ] I have checked the branch to which I would like to merge.

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description
Adopt ceil method for ios inapp prices i.e. pick closest higher price point for an inapp purchase. And if course price is > 1000 then don't apply price for this product. 
Previously for android purchases we were not storing transaction ids in paymentresponseextension table, this change will also fix this issue.
For user cancelled payments return user cancelled error in execute api view.

## Supporting information

Jira Ticket: https://2u-internal.atlassian.net/browse/LEARNER-10095

### CODE REVIEW

File: ecommerce/extensions/iap/api/v1/utils.py [hunk 1]
```diff
@@ -34,6 +34,9 @@ def create_ios_product(course, ios_product, configuration):
     Create in app ios product on connect store.
     return error message in case of failure.
     """
+    if course['price'] > 1000:
+        return 'Error: Appstore does not allow price> 1000'
```
Comment (christopappas) [inline ecommerce/extensions/iap/api/v1/utils.py]: nit: `price>` should be `price >`

File: ecommerce/extensions/iap/api/v1/utils.py [hunk 2]
```diff
@@ -188,15 +191,16 @@ def apply_price_of_inapp_purchase(price, in_app_purchase_id, headers):
...
 
-    nearest_low_price = nearest_low_price_id = 0
+    # Apple doesn't allow in app price > 1000
+    nearest_high_price = nearest_high_price_id = 1001
     for price_point in response.json()['data']:
         customer_price = float(price_point['attributes']['customerPrice'])
-        if nearest_low_price < customer_price <= price:
-            nearest_low_price = customer_price
-            nearest_low_price_id = price_point['id']
+        if nearest_high_price > customer_price >= price:
```
Comment (christopappas) [inline ecommerce/extensions/iap/api/v1/utils.py:198]: question: what is the scenario where `price` is greater than `customer_price`?

File: ecommerce/extensions/iap/api/v1/tests/test_utils.py [hunk 3]
```diff
@@ -73,17 +73,25 @@ def test_create_ios_product(self, _, __, ___, ____, _____, ______, _______):
...
 
-    # @mock.patch('ecommerce.extensions.iap.api.v1.utils.create_inapp_purchase')
+    def test_error_on_ios_product_price_threshhold(self, _,):
+        course = {
+            'key': 'test',
+            'name': 'test',
+            'price': 1001
+        }
+        error_msg = create_ios_product(course, self.ios_seat, self.configuration)
+        self.assertEqual(error_msg, 'Error: Appstore does not allow price> 1000')
```
Comment (christopappas) [inline ecommerce/extensions/iap/api/v1/tests/test_utils.py]: `price> 1000` should be `price > 1000`

File: ecommerce/extensions/iap/processors/base_iap.py [hunk 4]
```diff
@@ -212,6 +212,7 @@ def record_processor_response(self, response, transaction_id=None, basket=None,
                                                                             basket=basket)
 
         meta_data = self._get_metadata(currency_code=currency_code, price=price)
+        original_transaction_id = original_transaction_id or transaction_id
```
Comment (christopappas) [inline ecommerce/extensions/iap/processors/base_iap.py:215]: question: how does this change relate to the inapp price change?

File: ecommerce/extensions/iap/api/v1/utils.py (hunk 2 above)
Comment (jawad-khan) [reply ecommerce/extensions/iap/api/v1/utils.py:198]: customer price range is 0-1000 and we are already filtering > 1000 cases. Therefore there is no chance for this scenario. 
Even if it occurs this function will return exception and we will not apply price for that product and will investigate issue afterwards.

File: ecommerce/extensions/iap/processors/base_iap.py (hunk 4 above)
Comment (jawad-khan) [reply ecommerce/extensions/iap/processors/base_iap.py:215]: It doesn't. I noticed that nit and will be sending as a separate commit in this PR.

**Review by christopappas — APPROVED** [review]
good once CI goes ✅ 👍

**Pull request merged**